lib/index.js

```diff
@@ -57,7 +57,7 @@ internals.implementation = function (server, options) {
         cookieOptions.ttl = settings.ttl;
     }
 
-    if (settings.domain) {
+    if (settings.hasOwnProperty('domain')) {
```

Comment:
Sure, that makes sense. It's worth pointing out though, that with the validation happening first there's no way to trigger that branch of logic -- no falsey value will pass validation for domain, presently.